src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFirecloudGroupStep.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.model.FirecloudGroupModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class CreateSnapshotFirecloudGroupStep implements Step {
+  private final IamService iamService;
+  private final UUID snapshotId;
+
+  public CreateSnapshotFirecloudGroupStep(IamService iamService, UUID snapshotId) {
+    this.iamService = iamService;
+    this.snapshotId = snapshotId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FirecloudGroupModel created = iamService.createFirecloudGroup(snapshotId.toString());
```

Comment:
This will fail if the group already exists, but steps need to be idempotent so it shouldn't fail in that case.

Maybe it should be written as:
- does the group exist already?
- if not, create it
- otherwise, do nothing

Reply:
if the group already exists, it creates a new unique name and creates that group instead, not fail. This group will later be populated with the researcher and added as a dac, so I don't think we would want to modify an existing group.